opentimelineview/timeline_widget.py

```diff
@@ -338,34 +338,32 @@ def __init__(self, stack, *args, **kwargs):
         self._add_markers()
 
     def _adjust_scene_size(self):
-        # Make the scene large enough so that all items fit
-        all_ranges = [t.range_of_child_at_index(n)
-                      for t in self.stack for n in range(len(t))]
-
-        if all_ranges:
-            start_time = min(c.start_time for c in all_ranges)
-            end_time_exclusive = max(
-                c.end_time_exclusive() for c in all_ranges
-            )
-        else:
-            start_time = otio.opentime.RationalTime()
-            end_time_exclusive = otio.opentime.RationalTime()
+        scene_range = self.stack.trimmed_range()
 
-        start_time = otio.opentime.to_seconds(start_time)
-        duration = otio.opentime.to_seconds(end_time_exclusive)
+        start_time = otio.opentime.to_seconds(scene_range.start_time)
+        duration = otio.opentime.to_seconds(scene_range.end_time_exclusive())
 
-        has_video_tracks = any(
-            t for t in self.stack if t.kind == otio.schema.SequenceKind.Video
-        )
-        has_audio_tracks = any(
-            t for t in self.stack if t.kind == otio.schema.SequenceKind.Audio
-        )
+        if isinstance(self.stack, otio.schema.Stack):
+            has_video_tracks = any(
+                t.kind == otio.schema.SequenceKind.Video for t in self.stack
+            )
+            has_audio_tracks = any(
+                t.kind == otio.schema.SequenceKind.Audio for t in self.stack
+            )
```

Comment:
Would be great to show tracks with unknown types also. Silently ignoring them could be super confusing to people with odd track types.

Checking the track types is also a great thing for the validator to report.